src/EventObserver.js

```diff
@@ -26,12 +8,36 @@ class EventObserver {
   }
 
   unsubscribe (fn) {
-    this.observers = this.observers.filter(subscriber => subscriber !== fn);
+    this.observers = this.observers.filter(
+      subscriber => subscriber !== fn
+    );
   }
 
   broadcast (data) {
-    this.observers.forEach(subscriber => subscriber(data));
+    if(this.observers && this.observers.length > 0) {
+      this.observers.forEach(subscriber => subscriber(data));
+    }
   }
 }
 
-export default EventObserver;
\ No newline at end of file
+export default EventObserver;
+
+// you can implement observer even without classes!
+function functionalObserver() {
```

Comment:
if you don't want to use classes, you cant implement observer just with function